### PR TITLE
UUPS upgradeability

### PR DESCRIPTION
**Pull request by samigabor**
- StreamrConfig

### CODE REVIEW

**Review by jtakalai — APPROVED** [review]
I think we need to do another RBAC cleanup pass

File: packages/network-contracts/contracts/OperatorTokenomics/Sponsorship.sol [hunk 1]
```diff
@@ -148,10 +150,14 @@ contract Sponsorship is Initializable, ERC2771ContextUpgradeable, IERC677Receive
         metadata = metadata_;
         streamrConfig = globalStreamrConfig;
         __AccessControl_init();
+        __UUPSUpgradeable_init();
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender()); // factory needs this to set policies, (self-)revoke after policies are set!
+        _grantRole(UPGRADER_ROLE, _msgSender());
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/Sponsorship.sol:155]: updated in https://github.com/streamr-dev/network-contracts/pull/677

File: packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol [hunk 2]
```diff
@@ -129,13 +130,17 @@ contract StreamrConfig is Initializable, UUPSUpgradeable, AccessControlUpgradeab
...
 
-    // TODO: initializer arguments?
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() { _disableInitializers(); }
+
     function initialize() public initializer {
         __AccessControl_init();
         __UUPSUpgradeable_init();
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
         _setRoleAdmin(DEFAULT_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol:140]: updated in https://github.com/streamr-dev/network-contracts/pull/677

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipFactory.sol [hunk 3]
```diff
@@ -33,18 +34,19 @@ contract SponsorshipFactory is Initializable, UUPSUpgradeable, ERC2771ContextUpg
...
 
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor() ERC2771ContextUpgradeable(address(0x0)) {}
+    constructor() ERC2771ContextUpgradeable(address(0x0)) { _disableInitializers(); }
 
     function initialize(address templateAddress, address dataTokenAddress, address streamrConfigAddress) public initializer {
         __AccessControl_init();
+        __UUPSUpgradeable_init();
         _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _grantRole(UPGRADER_ROLE, _msgSender());
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/SponsorshipFactory.sol:43]: updated in https://github.com/streamr-dev/network-contracts/pull/677

File: packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol [hunk 4]
```diff
@@ -129,13 +130,17 @@ contract StreamrConfig is Initializable, UUPSUpgradeable, AccessControlUpgradeab
...
 
+        _grantRole(UPGRADER_ROLE, _msgSender());
+
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol:143]: updated in https://github.com/streamr-dev/network-contracts/pull/677

File: packages/network-contracts/contracts/OperatorTokenomics/Sponsorship.sol [hunk 5]
```diff
@@ -148,10 +150,14 @@ contract Sponsorship is Initializable, ERC2771ContextUpgradeable, IERC677Receive
         metadata = metadata_;
         streamrConfig = globalStreamrConfig;
         __AccessControl_init();
+        __UUPSUpgradeable_init();
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender()); // factory needs this to set policies, (self-)revoke after policies are set!
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/Sponsorship.sol:154]: fixed already by other PR

**Comment by jtakalai** [comment]
I think this was already fully accomplished in #677 

Good job 👍

**Pull request closed without merging**